### PR TITLE
Adds permissions filter for system service

### PR DESCRIPTION
**Pull request by evoevodin**
Allows to get state and stop system for those users who
have manage_codenvy permission.

Adapts code to changes made by eclipse/che#3917

### CODE REVIEW

**Comment by sleshchenko** [comment]
I don't see binding for [SystemEventsWebsocketBroadcaster](https://github.com/eclipse/che/pull/3917/files#diff-fc22f2d682ab2ded0594e97128a23880R185). It is correct?

File: wsmaster/codenvy-hosted-system/src/main/java/com/codenvy/service/system/SystemServicePermissionsFilter.java [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+    protected void filter(GenericResourceMethod resource, Object[] args) throws ApiException {
+        switch (resource.getMethod().getName()) {
+            case "stop":
+            case "getState":
+                EnvironmentContext.getCurrent()
+                                  .getSubject()
+                                  .checkPermission(SystemDomain.DOMAIN_ID, null, SystemDomain.MANAGE_SYSTEM_ACTION);
+                break;
+            default:
+                // all the other methods may be called by anyone
```
Comment (sleshchenko) [inline wsmaster/codenvy-hosted-system/src/main/java/com/codenvy/service/system/SystemServicePermissionsFilter.java]: I'd list all public methods and throw ForbiddenException by default.

Comment (evoevodin) [inline wsmaster/codenvy-hosted-system/src/main/java/com/codenvy/service/system/SystemServicePermissionsFilter.java]: ok

File: wsmaster/codenvy-hosted-system/src/test/java/com/codenvy/service/system/SystemServicePermissionsFilterTest.java [hunk 2]
```diff
@@ -0,0 +1,190 @@
...
+
+    @Mock
+    private static Subject subject;
+
+    @Mock
+    private HostedSystemService systemService;
+
+    @BeforeMethod
+    public void setUp() throws Exception {
+        when(subject.getUserId()).thenReturn(USER_ID);
```
Comment (sleshchenko) [inline wsmaster/codenvy-hosted-system/src/test/java/com/codenvy/service/system/SystemServicePermissionsFilterTest.java]: Looks like you can remove USER_ID at all.

File: wsmaster/codenvy-hosted-system/src/test/java/com/codenvy/service/system/SystemServicePermissionsFilterTest.java [hunk 3]
```diff
@@ -0,0 +1,190 @@
...
+               .when()
+               .post(SECURE_PATH + "/system/stop")
+               .then()
+               .statusCode(204);
+
+        verify(systemService).stop();
+    }
+
+    @Test
+    public void rejectsRequestForUserWithoutManageSystemPermission() throws Exception {
```
Comment (sleshchenko) [inline wsmaster/codenvy-hosted-system/src/test/java/com/codenvy/service/system/SystemServicePermissionsFilterTest.java]: `rejectsRequestForUserWithoutManageSystemPermission` -> `rejectsStopForUserWithoutManageSystemPermission`

**Pull request merged**